src/crawlers/cheerio_crawler.js

```diff
@@ -464,24 +473,32 @@ class CheerioCrawler {
     /**
      * Wrapper around handlePageFunction that opens and closes pages etc.
      *
-     * @param {Object} options
-     * @param {Request} options.request
-     * @param {AutoscaledPool} options.autoscaledPool
-     * @param {Session} [options.session]
+     * @param {Object} crawlingContext
+     * @param {Request} crawlingContext.request
+     * @param {AutoscaledPool} crawlingContext.autoscaledPool
+     * @param {Session} [crawlingContext.session]
      * @ignore
      */
-    async _handleRequestFunction({ request, autoscaledPool, session }) {
-        let proxyInfo;
+    async _handleRequestFunction(crawlingContext) {
         let proxyUrl;
+        const { request, session } = crawlingContext;
+
         if (this.proxyConfiguration) {
-            proxyInfo = this.proxyConfiguration.newProxyInfo(session ? session.id : undefined);
-            proxyUrl = proxyInfo.url;
+            const sessionId = session ? session.id : undefined;
+            crawlingContext.proxyInfo = this.proxyConfiguration.newProxyInfo(sessionId);
+            proxyUrl = crawlingContext.proxyInfo.url;
         }
 
-        if (this.prepareRequestFunction) await this.prepareRequestFunction({ request, session, proxyInfo });
+        if (this.prepareRequestFunction) await this.prepareRequestFunction(crawlingContext);
 
         const { dom, isXml, body, contentType, response } = await addTimeoutToPromise(
-            this._requestFunction({ request, session, proxyUrl }),
+            this._requestFunction(
+                {
+                    request,
+                    session,
+                    proxyUrl,
+                },
+            ),
```

Comment:
please squash this back to a single line and do `proxyUrl: crawlingContext.proxyInfo.url`